src/groups/posts.ts

```diff
@@ -0,0 +1,76 @@
+import db from '../database';
+import privileges from '../privileges';
+import groups from '.';
+import posts from '../posts';
+import topics from '../topics';
+import { PostObject } from '../types/post';
+import { GroupDataObject, TopicObject } from '../types';
+
+interface GroupsInterface {
+  onNewPostMade: (postData: { uid: number, cid: number, pid: number, timestamp: string }) => Promise<void>;
+  getLatestMemberPosts: (groupName: string, max: number, uid: string) => Promise<PostObject[]>;
+  getUserGroupMembership: (set: string, uids: number[]) => Promise<string[][]>;
+}
+
+export = function (Groups: GroupsInterface) {
+    async function truncateMemberPosts(groupName: string) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+        const pids = await db.getSortedSetRevRange(`group:${groupName}:member:pids`, 10, 10) as number[];
+        const lastPid = pids[0];
```

Comment:
Does this need to be explicitly typeset as a `number`

Reply:
Yea, the type here in inferred since `pids` is of type `number[]`